civihr_default_theme/template.php

```diff
@@ -355,3 +355,63 @@ class="col-sm-3 control-label ' . ( $label_hidden ? 'hidden-xs' : '' ) . '">'
   return $fields_structure;
 }
 
+/**
+ * Returns HTML for a menu link and submenu.
+ *
+ * @param $variables
+ *   An associative array containing:
+ *   - element: Structured array data for a menu link.
+ *
+ * @ingroup themeable
+ */
+function civihr_default_theme_menu_link__dropdown($variables)
+{
+    $element = $variables['element'];
+    $sub_menu = '';
+
+    if (!empty($element['#below'])) {
+        // Wrap in dropdown-menu.
+        unset($element['#below']['#theme_wrappers']);
+        $sub_menu = '<ul class="dropdown-menu">' . drupal_render($element['#below']) . '</ul>';
+        $element['#localized_options']['attributes']['class'][] = 'dropdown-toggle';
+        $element['#localized_options']['attributes']['data-toggle'] = 'dropdown';
+
+        // Check if element is nested.
+        if ((!empty($element['#original_link']['depth'])) && ($element['#original_link']['depth'] > 1)) {
+            $element['#attributes']['class'][] = 'dropdown-submenu';
+        }
+        else {
+            $element['#attributes']['class'][] = 'dropdown';
+            $element['#localized_options']['html'] = TRUE;
+            $element['#title'] .= '<span class="caret"></span>';
+        }
+
+        $element['#localized_options']['attributes']['data-target'] = '#';
+    }
+
+    // Fix for active class.
+    if (($element['#href'] == current_path() || ($element['#href'] == '<front>' && drupal_is_front_page())) && (empty($element['#localized_options']['language']) || $element['#localized_options']['language']->language == $language_url->language)) {
+        $element['#attributes']['class'][] = 'active';
+    }
+
+    // Add active class to li if active trail.
+    if (in_array('active-trail', $element['#attributes']['class'])) {
+        $element['#attributes']['class'][] = 'active';
+    }
+
+    // Add a unique class using the title.
+    $title = strip_tags($element['#title']);
+    $element['#attributes']['class'][] = 'menu-link-' . drupal_html_class($title);
+
+    /* Code Added */
+    //Hide the links to civicrm and dashboard if Access to CiviCRM is not present
```

Comment:
If we really need to copy paste the entire `menu_link__dropdown` function (and again, I want to make sure that this is the *very* last resort), then at least put your "code added" stuff in a separate function